### ISSUE TITLE
Display Error when using data labels and min and/or max values for the x-axis

### ISSUE THREAD

**Issue by team-moeller** [team-moeller/better-access-charts#170]
When you decide to show data labels and also decide to use min and/or max values for the x-axis there is an error in the display.

**Comment by team-moeller** [comment]
This is the original chart without min and max values for the x-axis:
![image](https://github.com/team-moeller/better-access-charts/assets/74451434/fc83c799-2317-4bdb-8a1b-1cdbbbf24fbf)
The datalables are shown at the beginning and the color is red.

**Comment by team-moeller** [comment]
This is the chart with values for min and max for the x-axis:
![image](https://github.com/team-moeller/better-access-charts/assets/74451434/fc5270aa-ad36-4a7c-94c2-412baf23fab8)
One can see the number "3" at the left side and the numbers 6, 9 and 4 on the right side.

**Comment by team-moeller** [comment]
When we open the underlying html-code directly in the browser the problem also exists:
![image](https://github.com/team-moeller/better-access-charts/assets/74451434/c72fce17-8b27-443a-9a0e-8e6377ba35b6)

**Comment by team-moeller** [comment]
This issue was reported to the data labels team:
https://github.com/chartjs/chartjs-plugin-datalabels/issues/414

**Comment by team-moeller** [comment]
The user [@yannick_xy](https://github.com/Yannick-XY) brought my attention to this video: https://www.youtube.com/watch?v=pbcH5H3mNd4

At minute 09:31 the optimized code is visible: https://www.youtube.com/watch?v=pbcH5H3mNd4&t=591s

**Comment by team-moeller** [comment]
With the help of [@yannick_xy](https://github.com/Yannick-XY) I got the above solution adapted to better Access Charts.

Old Code:
```html
datalabels: {
    anchor: 'start',
    color: 'red',
    font: {
        size: 16
    }
}
```
New Code:

```html
datalabels: {
    anchor: 'start',
    color: (context) => {
        const x = context.chart.scales.x;
        if (context.dataIndex >= x.min && context.dataIndex <= x.max) {
            return 'red';
        } else {
            return 'transparent';
        }
    }
}
```

**Comment by team-moeller** [comment]
Released with version [4.10.04](https://github.com/team-moeller/better-access-charts/releases/tag/4.10.04)
